tests/test_bar_data.py

```diff
@@ -476,6 +476,38 @@ def test_is_stale_at_midnight(self):
         with handle_non_market_minutes(bar_data):
             self.assertTrue(bar_data.is_stale(self.HILARIOUSLY_ILLIQUID_ASSET))
 
+    def test_overnight_adjustments(self):
+        # verify there is a split for SPLIT_ASSET
+        splits = self.adjustments_reader.get_adjustments_for_sid(
+            "splits",
+            self.SPLIT_ASSET.sid
+        )
+
+        self.assertEqual(1, len(splits))
+        split = splits[0]
+        self.assertEqual(
+            split[0],
+            pd.Timestamp("2016-01-06", tz='UTC')
+        )
+
+        # Current day is 1/06/16
+        day = self.days[1]
+        eight_fortyfive_am_eastern = \
+            pd.Timestamp("{0}-{1}-{2} 8:45".format(
+                day.year, day.month, day.day),
+                tz='US/Eastern'
+            )
+
+        bar_data = BarData(self.data_portal,
+                           lambda: eight_fortyfive_am_eastern,
+                           "minute")
+
+        with handle_non_market_minutes(bar_data):
```

Comment:
I think you need to test with every possible field value (OHLCVP), and verify that for OHLCP, the values are multiplied by the split ratio, and for V, the value is divided by the split ratio.